File: BC-W4/Army/Spells/DragonFlame/DragonFlame.cpp
#include "DragonFlame.h"

DragonFlame::DragonFlame(int actionPoints, int manaCost, const std::string& spellName) : Spell(actionPoints, manaCost, spellName) {};
DragonFlame::~DragonFlame() {};

void DragonFlame::action(Unit* enemy) {
    enemy->takeMagicDamage(this->getActionPoints());
};

void DragonFlame::heal(Unit* unit) {
    throw InvalidHealException();
};

File: BC-W4/ArmyJava/src/spells/DragonFlame.java
package spells;

import exceptions.InvalidCastException;
import units.Unit;

public class DragonFlame extends Spell {
    public DragonFlame() {
        super("DragonFlame", 55, 80, true);
    }

    public DragonFlame(int actionPoints) {
        super("DragonFlame", 55, actionPoints, true);
    }

    public void action(Unit enemy) throws InvalidCastException {
        enemy.takeMagicDamage(this.actionPoints);
    }

}
